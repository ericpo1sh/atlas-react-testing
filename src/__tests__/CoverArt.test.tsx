import { render, screen } from "@testing-library/react";
import { expect, test } from "vitest";
import CoverArt from "../components/CoverArt";
import { MusicContext } from "../Context/MusicContext";


test("Cover Art Renders Correctly On Fail", () => {
  const { container } = render(<CoverArt/>);
  expect(container).toMatchSnapshot();
});

const mockContextValue = {
  CurrentSong: {
    cover: "https://m.media-amazon.com/images/I/71pxGj4RoVS._AC_UF894,1000_QL80_.jpg",
    title: "Graduation",
  },
};

const mockEmptyContext = null;

test("Matches snapshot when CurrentSong is available", () => {
  const { container } = render(
    <MusicContext.Provider value={mockContextValue}>
      <CoverArt />
    </MusicContext.Provider>
  );

  expect(container).toMatchSnapshot();
});
