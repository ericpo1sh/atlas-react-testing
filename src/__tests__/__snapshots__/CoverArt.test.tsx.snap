// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Cover Art Renders Correctly 1`] = `
<div>
  <div>
    Not Found
  </div>
</div>
`;

exports[`Cover Art Renders Correctly On Fail 1`] = `
<div>
  <div>
    Not Found
  </div>
</div>
`;

exports[`Matches snapshot when CurrentSong is available 1`] = `
<div>
  <div
    class="flex items-center justify-center"
  >
    <img
      alt="Cover Art"
      class="rounded-2xl w-96 h-96"
      src="https://m.media-amazon.com/images/I/71pxGj4RoVS._AC_UF894,1000_QL80_.jpg"
    />
  </div>
</div>
`;
